src/blocking/mod.rs

```diff
@@ -229,6 +229,7 @@ mod test {
     }
 
     #[test]
+    #[ignore]
```

Comment:
Do these tests not work in CI because they require some kind of interaction to set the password for newly created collections? If so, it would be nice if we dynamically ignored them by looking for a GitHub CI environment variable so that they can still run on developer's local machines.

Reply:
>Do these tests not work in CI because they require some kind of interaction to set the password for newly created collections? 

Precisely.

>If so, it would be nice if we dynamically ignored them by looking for a GitHub CI environment variable so that they can still run on developer's local machines.

Github Actions sets the environment variable `GITHUB_ACTIONS`, so that would be the best way to detect whether we're running via CI or not. However, rust currently doesn't offer a way to ignore tests based on environment variables, it's an open feature request: https://github.com/rust-lang/rust/issues/68007
Workarounds are discussed here: https://stackoverflow.com/questions/73760692/ignore-a-test-based-on-environment-variable

IMO adding a new feature and then setting that feature in the CI script would be the best workaround.

Reply:
Since gnome-keyring uses `gcr-prompter`, available on dbus via `org.gnome.keyring.SystemPrompter`, another possibility would be to create a dummy service that acts as `org.gnome.keyring.SystemPrompter` and answers any prompts with an empty password.
This way even the currently ignored tests can be run via CI. However, I'm not sure if it's worth the effort.

Reply:
> Github Actions sets the environment variable GITHUB_ACTIONS, so that would be the best way to detect whether we're running via CI or not. However, rust currently doesn't offer a way to ignore tests based on environment variables

Since there are only two tests affected, could we just add a guard in each like this?
```rust
fn is_running_ci() -> bool {
    std::env::var("GITHUB_ACTIONS").is_ok()
}

#[test]
fn should_create_and_delete_collection() {
    if is_running_ci() {
        return;
    }
}

...
```

> However, I'm not sure if it's worth the effort.

I agree 👍. This seems like too much work for two tests.

Reply:
>Since there are only two tests affected, could we just add a guard in each like this?

Yep, that's also possible. The disadvantage is that it shows the tests as `ok`, misleadingly indicating that the tests succeeded despite them actually not being executed. But since this solution is simple and doesn't require any new features in `Cargo.toml`, I implemented it the new commit I just pushed.

Reply:
Hi there,
There is a crate that can help you with the issue with `ok` in ignore test case.
https://crates.io/crates/test-with
Just for reference. :smile:

Reply:
@yanganto Thanks for your comment. I also stumbled across your crate while looking for a solution. It would make it possible to ignore tests if an environment variable is not set. But does it also allow you to ignore tests if an environment variable is set? I haven't found anything regarding that in the docs.

Reply:
Sorry, currently use case is not covered for ignoring when the environment variable is set. I will add a feature for this use case over the weekend if you have time to wait on this.

Reply:
@complexspaces What's your opinion on using this crate to ignore tests?

Reply:
I would be fine using it since its just a dev dependency and it seems like it will do exactly what we want 👍

Reply:
Hi, 
This is the crate with the correct version and minimal features for this scenario.
`test-with = { version = "0.8", default-features = false }`

And this is the macro for this scenario.
`#[test_with::no_env(GITHUB_ACTIONS)]`

:smile:

Reply:
Alright, implemented as discussed. If `GITHUB_ACTIONS` is set at compile-time, the respective tests won't be executed.